### PR TITLE
add stubs to some modules

### PR DESCRIPTION
**Pull request by nvnieuwk**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/gatk4/haplotypecaller/main.nf [hunk 1]
```diff
@@ -55,4 +55,21 @@ process GATK4_HAPLOTYPECALLER {
         gatk4: \$(echo \$(gatk --version 2>&1) | sed 's/^.*(GATK) v//; s/ .*\$//')
     END_VERSIONS
     """
+
+    stub:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def bamout_command = args.contains("--bam-writer-type") ? "--bam-output ${prefix.replaceAll('.g\\s*$', '')}.realigned.bam" : ""
+
+    def stub_realigned_bam = bamout_command ? "touch ${prefix.replaceAll('.g\\s*$', '')}.realigned.bam" : ""
```
[the remark this answers is not in the document]
Comment (nvnieuwk) [reply modules/nf-core/gatk4/haplotypecaller/main.nf:64]: I've tried to stay as close to the original code as possible. That's why I didn't merge them :)

[the remark this answers is not in the document]
Comment (JoseEspinosa) [reply modules/nf-core/gatk4/haplotypecaller/main.nf:64]: Fair enough! that is why I put a `?`, I thought that probably that was the reason but still suggested simplifying the code, but it's ok :smile:

**Review by JoseEspinosa — APPROVED** [review]
Awesome! :rocket:

File: modules/nf-core/gatk4/haplotypecaller/main.nf (hunk 1 above)
[the remark this answers is not in the document]
Comment (nvnieuwk) [reply modules/nf-core/gatk4/haplotypecaller/main.nf:64]: Ah but it doesn't really matter to me tbh, I just thought that was the guideline on stubs :) Would you prefer it if they were merged?

**Pull request merged**